src/test/java/com/iota/iri/network/impl/TransactionRequesterWorkerImplTest.java

```diff
@@ -0,0 +1,153 @@
+package com.iota.iri.network.impl;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import com.iota.iri.TangleMockUtils;
+import com.iota.iri.controllers.TipsViewModel;
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.model.persistables.Transaction;
+import com.iota.iri.network.Node;
+import com.iota.iri.network.TransactionRequester;
+import com.iota.iri.network.impl.TransactionRequesterWorkerImpl;
+import com.iota.iri.service.snapshot.SnapshotProvider;
+import com.iota.iri.storage.Tangle;
+import com.iota.iri.zmq.MessageQ;
+
+import static com.iota.iri.TransactionTestUtils.getRandomTransaction;
+import static com.iota.iri.TransactionTestUtils.get9Transaction;
+import static com.iota.iri.TransactionTestUtils.buildTransaction;
+import static com.iota.iri.TransactionTestUtils.getRandomTransactionHash;
+
+import static org.mockito.Mockito.when;
+
+public class TransactionRequesterWorkerImplTest {
+    
+    //Good
+    private static final TransactionViewModel TVMRandomNull = new TransactionViewModel(
+            getRandomTransaction(), Hash.NULL_HASH);
+    private static final TransactionViewModel TVMRandomNotNull = new TransactionViewModel(
+            getRandomTransaction(), getRandomTransactionHash());
+    private static final TransactionViewModel TVMAll9Null = new TransactionViewModel(
+            get9Transaction(), Hash.NULL_HASH);
+    private static final TransactionViewModel TVMAll9NotNull = new TransactionViewModel(
+            get9Transaction(), getRandomTransactionHash()); 
+    
+    //Bad
+    private static final TransactionViewModel TVMNullNull = new TransactionViewModel((Transaction)null, Hash.NULL_HASH); 
+    
+    @Rule 
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+    
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private static SnapshotProvider snapshotProvider;
+    
+    @Mock
+    private static MessageQ messageQ;
+    
+    private static TransactionRequester requester;
+    private static TransactionRequesterWorkerImpl worker;
+
+    @Mock
+    private Tangle tangle;
+    
+    @Mock
+    private Node node;
+    
+    @Mock
+    private TipsViewModel tipsVM;
+
+    @Before
+    public void before() throws NoSuchFieldException, SecurityException, Exception {
```

Comment:
No need to declare `throws` :-)
When you commit there should be an option in the IDE to perform code analysis